src/extension/importer/core/src/main/java/org/geoserver/importer/TransformingCancelableInteratingFC.java

```diff
@@ -0,0 +1,150 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.importer;
+
+import java.util.NoSuchElementException;
+
+import org.geoserver.importer.job.ProgressMonitor;
+import org.geoserver.importer.transform.VectorTransformChain;
+import org.geotools.data.DataStore;
+import org.geotools.feature.FeatureCollection;
+import org.geotools.feature.FeatureIterator;
+import org.geotools.feature.collection.DecoratingFeatureCollection;
+import org.geotools.feature.collection.DecoratingFeatureIterator;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.opengis.feature.Feature;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.FeatureType;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+/**
+ * FeatureCollection that does two things required by the importer; a) provide cancel functionality b) Do some FeatureType Transforming
+ *
+ * This class is simply wraps the FeatureIterator with two iterators wrappers that provide the above functionality.
+ */
+public class TransformingCancelableInteratingFC extends DecoratingFeatureCollection {
```

Comment:
Nice, you found DecoratingFeatureCollection. 

- You may wish to make this package visible, rather than commit to additional public api
- Can we rename this to "ImportTransformFeatureCollection" or something to follow the naming convention.